resultsdbupdater/utils.py

```diff
@@ -64,6 +64,54 @@ def create_result(log, testcase, outcome, ref_url, data, groups=None, note=None)
     post_req.raise_for_status()
 
 
+def update_common_ci_data(msg, outcome, data):
+    data.update(msg.contact_dict)
+    data['recipients'] = msg.recipients
+
+    update_publisher_id(data=data, msg=msg)
+
+    if outcome == 'ERROR':
+        data['error_reason'] = msg.error_reason[:256]
+
+        issue_url = msg.get('error', 'issue_url', default=None)
+        if issue_url:
+            data['issue_url'] = issue_url
+
+
+def product_scenario_builds(product_scenario, products):
+    builds_data = []
+    for product in products:
+        if product.get('type') != 'product-build':
+            continue
+
+        try:
+            build_data = {
+                'product_scenario': product_scenario,
+
+                # required values
+                'type': product['type'],
+                'item': product['id'],
```

Comment:
Hmm... I think the [specification](https://pagure.io/fedora-ci/messages/blob/master/f/schemas/product-build.yaml) is not very clear here. It says `ID of the product meaning the sha256 digest of the sorted ids of the artifacts list`.

However, the artifacts list is defined as `List of child artifacts, from which this product consists. It is required, but can be an empty list.`

So what will be the `id` if the list is empty?

I will open an issue for the specification.

Reply:
Maybe let's try something different here.

Can we take `product['nvr']` instead of `product['id']`? That way we can bypass the unclear part of the specification.

Reply:
I could take `nvr` if available (since it's optional), otherwise fall back to `id`.